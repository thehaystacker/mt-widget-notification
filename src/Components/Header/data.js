export default {
  data: {
    1010: {
      identifier: 1010,
      username: "Dan Auerbach",
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing",
      created: "15 September 2021",
      prof_ico: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0A816QQ7i-w2DLwWBdY0nmq8-lHCUlcUHBw&usqp=CAU',
      is_read: true,
    },
    1011: {
      identifier: 1011,
      username: "Mia Brian",
      text: "sed do eiusmod tempor incididunt ut labore et dolore",
      created: "8 October 2021",
      prof_ico: 'https://i0.wp.com/worldbusinessfitness.com/wp-content/uploads/2018/06/opulent-profile-square-02.jpg',
      is_read: false,
    },
    1012: {
      identifier: 1012,
      username: "Emily Broemel",
      text: "minim veniam, quis nostrud exercitation",
      created: "20 October 2021",
      prof_ico: 'https://www.nzherald.co.nz/resizer/qsciwSumABgKMvLjKYvh2TK1sGs=/360x384/filters:quality(70)/arc-anglerfish-syd-prod-nzme.s3.amazonaws.com/public/QZI5SNWLCJANLBVQHUDCEQTNTE.jpg',
      is_read: false,
    },
    1013: {
      identifier: 1013,
      username: "Dan Auerbach",
      text: "Excepteur sint occaecat cupidatat non proident",
      created: "18 April 2012",
      prof_ico: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQGfF_3ycXBYHLIC4NWYcovKN8W_dsy98dqQw&usqp=CAU',
      is_read: true,
    },
    1014: {
      identifier: 1014,
      username: "Alex Lifeson",
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing",
      created: "15 September 2021",
      prof_ico: 'https://uilove.in/realestate/listo/preview/img/demo/profile.jpg',
      is_read: false,
    },
    1015: {
      identifier: 1010,
      username: "Dan Auerbach",
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing",
      created: "15 September 2021",
      prof_ico: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0A816QQ7i-w2DLwWBdY0nmq8-lHCUlcUHBw&usqp=CAU',
      is_read: true,
    },
    1016: {
      identifier: 1011,
      username: "Mia Brian",
      text: "sed do eiusmod tempor incididunt ut labore et dolore",
      created: "8 October 2021",
      prof_ico: 'https://i0.wp.com/worldbusinessfitness.com/wp-content/uploads/2018/06/opulent-profile-square-02.jpg',
      is_read: true,
    },
    1017: {
      identifier: 1012,
      username: "Carl Broemel",
      text: "minim veniam, quis nostrud exercitation",
      created: "20 October 2021",
      prof_ico: 'https://www.eventiz.biz/wp-content/uploads/2015/12/speaker-3.jpg',
      is_read: true,
    },
    1018: {
      identifier: 1013,
      username: "Dan Auerbach",
      text: "Excepteur sint occaecat cupidatat non proident",
      created: "18 April 2012",
      prof_ico: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQGfF_3ycXBYHLIC4NWYcovKN8W_dsy98dqQw&usqp=CAU',
      is_read: true,
    },
    1019: {
      identifier: 1014,
      username: "Alex Lifeson",
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing",
      created: "15 September 2021",
      prof_ico: 'https://uilove.in/realestate/listo/preview/img/demo/profile.jpg',
      is_read: false,
    },
    1020: {
      identifier: 1010,
      username: "Dan Auerbach",
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing",
      created: "15 September 2021",
      prof_ico: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0A816QQ7i-w2DLwWBdY0nmq8-lHCUlcUHBw&usqp=CAU',
      is_read: false,
    },
    1021: {
      identifier: 1011,
      username: "Mia Brian",
      text: "sed do eiusmod tempor incididunt ut labore et dolore",
      created: "8 October 2021",
      prof_ico: 'https://i0.wp.com/worldbusinessfitness.com/wp-content/uploads/2018/06/opulent-profile-square-02.jpg',
      is_read: true,
    },
    1022: {
      identifier: 1012,
      username: "Carl Broemel",
      text: "minim veniam, quis nostrud exercitation",
      created: "20 October 2021",
      prof_ico: 'https://www.eventiz.biz/wp-content/uploads/2015/12/speaker-3.jpg',
      is_read: true,
    },
    1023: {
      identifier: 1013,
      username: "Dan Auerbach",
      text: "Excepteur sint occaecat cupidatat non proident",
      created: "18 April 2012",
      prof_ico: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQGfF_3ycXBYHLIC4NWYcovKN8W_dsy98dqQw&usqp=CAU',
      is_read: true,
    },
    1024: {
      identifier: 1014,
      username: "Alex Lifeson",
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing",
      created: "15 September 2021",
      prof_ico: 'https://uilove.in/realestate/listo/preview/img/demo/profile.jpg',
      is_read: true,
    },
  },
  success: true,
  unread_count: 5,
  total: 125,
  limit: 15,
};
