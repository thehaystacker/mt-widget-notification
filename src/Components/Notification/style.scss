@import '../../Assets/scss/variables';

.wrap-notif {
  .wrap-ico {
    cursor: pointer;
    position: relative;

    .wrap-badge {
      position: absolute;
      font-size: 9px;
      font-weight: 600;
      top: -5px;
      left: 10px;
      background-color: $primary;
      color: #fff;
      padding: 2px 4px;
      border-radius: 10px;
    }
  }

  .wrap-dropdown {
    position: relative;
    display: none;

    .list-box {
      background-color: #fff;
      width: 350px;
      position: absolute;
      box-shadow: 0px 0px 10px 2px #e8ebf3;
      border-radius: 8px;
      left: 50%;
      transform: translateX(-50%);
      margin-top: 20px;
      z-index: 3;

      &::before {
        content: '';
        position: absolute;
        width: 10px;
        height: 10px;
        background-color: white;
        top: -5px;
        left: 50%;
        transform: translateX(-50%) rotate(45deg);
        box-shadow: 0px 0px 10px 2px #e8ebf3;
        z-index: 1;
      }

      &::after {
        content: '';
        position: absolute;
        width: 20px;
        height: 20px;
        background-color: #ffffff;
        top: 0px;
        left: 50%;
        transform: translateX(-50%) rotate(0);
        z-index: 2;
      }

      .header {
        padding: 20px;
        display: flex;
        justify-content: space-between;
        border-bottom: 1px solid #e8ebf3;

        .ico-header {
          font-size: 16px;

          &.bell {
            color: $primary;
          }

          &.read-all {
            color: $grey;
          }
        }
      }

      .footer {
        padding: 20px;
        text-align: center;
      }
    }

    .overlay {
      position: fixed;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-color: transparent;
      z-index: 2;
    }
  }

  &.open {
    .wrap-dropdown {
      display: block;
    }
  }

}