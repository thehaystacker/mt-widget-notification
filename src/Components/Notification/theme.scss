// Primary
.wrap-notif {
  &.primary {
    .wrap-ico {
      .wrap-badge {
        background-color: $primary;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $primary;
          }
        }
      }

      .footer {
        a {
          color: $primary;
        }
      }
    }

    .list {
      .item {
        &.unread {
          background-color: rgba($primary, 0.05);
        }
      }
    }
  }
}

// Success
.wrap-notif {
  &.success {
    .wrap-ico {
      .wrap-badge {
        background-color: $success;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $success;
          }
        }
      }

      .footer {
        a {
          color: $success;
        }
      }
    }

    .list {
      .item {
        &.unread {
          background-color: rgba($success, 0.05);
        }
      }
    }
  }
}

// Warning
.wrap-notif {
  &.warning {
    .wrap-ico {
      .wrap-badge {
        background-color: $warning;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $warning;
          }
        }
      }

      .footer {
        a {
          color: $warning;
        }
      }
    }

    .list {
      .item {
        &.unread {
          background-color: rgba($warning, 0.05);
        }
      }
    }
  }
}

// Success
.wrap-notif {
  &.success {
    .wrap-ico {
      .wrap-badge {
        background-color: $success;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $success;
          }
        }
      }

      .footer {
        a {
          color: $success;
        }
      }
    }

    .list {
      .item {
        &.unread {
          background-color: rgba($success, 0.05);
        }
      }
    }
  }
}

// Danger
.wrap-notif {
  &.danger {
    .wrap-ico {
      .wrap-badge {
        background-color: $danger;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $danger;
          }
        }
      }

      .footer {
        a {
          color: $danger;
        }
      }
    }

    .list {
      .item {
        &.unread {
          background-color: rgba($danger, 0.05);
        }
      }
    }
  }
}

// .wrap-notif {
//   .wrap-ico {
//     .wrap-badge {
//       background-color: $primary;
//       color: $white;
//     }
//   }

//   .wrap-dropdown {
//     .list-box {
//       background-color: $white;
//       box-shadow: 0px 0px 10px 2px $light-grey;

//       &::before {
//         background-color: $white;
//         box-shadow: 0px 0px 10px 2px $light-grey;
//       }

//       &::after {
//         background-color: $white;
//       }

//       .header {
//         .ico-header {
//           &.bell {
//             color: $primary;
//           }
//         }
//       }

//       .footer {
//         a {
//           color: $primary;
//         }
//       }
//     }
//   }

//   .list {
//     .item {
//       &.unread {
//         background-color: rgba($primary, 0.05);
//       }

//       .content {
//         color: $dark-grey;
//       }
//     }
//   }
// }