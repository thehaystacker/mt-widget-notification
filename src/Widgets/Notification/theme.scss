// Primary
.wrap-notif {
  &.primary {
    .wrap-ico {
      .wrap-badge {
        background-color: $primary;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $primary;
          }
        }
      }

      .footer {
        a {
          color: $primary;
        }
      }
    }

    .list {
      a {
        &:hover {
          background-color: rgba($color: $primary, $alpha: 0.05);
        }
      }

      .item {
        &.unread {
          &::before {
            background: $primary;
          }
        }
      }
    }
  }
}

// Success
.wrap-notif {
  &.success {
    .wrap-ico {
      .wrap-badge {
        background-color: $success;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $success;
          }
        }
      }

      .footer {
        a {
          color: $success;
        }
      }
    }

    .list {
      a {
        &:hover {
          background-color: rgba($color: $success, $alpha: 0.05);
        }
      }

      .item {
        &.unread {
          &::before {
            background: $success;
          }
        }
      }
    }
  }
}

// Warning
.wrap-notif {
  &.warning {
    .wrap-ico {
      .wrap-badge {
        background-color: $warning;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $warning;
          }
        }
      }

      .footer {
        a {
          color: $warning;
        }
      }
    }

    .list {
      a {
        &:hover {
          background-color: rgba($color: $warning, $alpha: 0.05);
        }
      }

      .item {
        &.unread {
          &::before {
            background: $warning;
          }
        }
      }
    }
  }
}

// Danger
.wrap-notif {
  &.danger {
    .wrap-ico {
      .wrap-badge {
        background-color: $danger;
      }
    }

    .wrap-dropdown {
      .header {
        .ico-header {
          &.bell {
            color: $danger;
          }
        }
      }

      .footer {
        a {
          color: $danger;
        }
      }
    }

    .list {
      a {
        &:hover {
          background-color: rgba($color: $danger, $alpha: 0.05);
        }
      }

      .item {
        &.unread {
          &::before {
            background: $danger;
          }
        }
      }
    }
  }
}

.wrap-notif {
  &.dark {
    .wrap-dropdown {
      .list-box {
        color: $white;
        background-color: $black;
        box-shadow: 0px 0px 10px 2px $grey;

        .header {
          border-bottom: 1px solid $dark-grey;
        }

        .footer {
          border-top: 1px solid $dark-grey;
        }

        &::before {
          background-color: $black;
          box-shadow: 0px 0px 10px 2px $grey;
        }

        &::after {
          background-color: $black;
        }
      }
    }

    .list {
      a {
        border-bottom: 1px solid $dark-grey;
      }

      .item {
        .content {
          color: $white;
        }
      }
    }
  }
}