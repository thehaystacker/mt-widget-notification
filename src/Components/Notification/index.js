import React, { useState } from "react";
import PropTypes from "prop-types";
import { faBell, faCheckCircle } from "@fortawesome/free-regular-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Link } from "react-router-dom";

import "./style.scss";
import Notifications from "../../Pages/Notifications";

function Notification(props) {
  // destructuring the props
  const { data, theme, variant, markAllRead } = props;
  // component local state
  const [open, setOpen] = useState(false);

  return (
    <div
      className={[`wrap-notif`, theme, variant, open ? `open` : ``].join(" ")}
    >
      <div className="wrap-ico" onClick={() => setOpen(true)}>
        <FontAwesomeIcon icon={faBell} className="ico" />
        <div className="wrap-badge">{data.unread_count}</div>
      </div>

      <div className="wrap-dropdown">
        <div className="list-box">
          <div className="header">
            <FontAwesomeIcon icon={faBell} className="ico-header bell" />
            <h5 className="txt-h5">Notifications</h5>
            <FontAwesomeIcon
              icon={faCheckCircle}
              className="ico-header read-all link-ico"
              onClick={markAllRead}
              data-testid="btn-markallread"
              title="Mark all as read"
            />
          </div>

          {data.data ? (
            <div className="list">
              {Object.keys(data.data).map((id) => (
                <Link to={`/notifications/${id}`} target="_blank" rel="noopener noreferrer" key={id}>
                  <div
                    className={[
                      `item`,
                      data.data[id].is_read ? `` : `unread`,
                    ].join(" ")}
                  >
                    <div className="prof-ico">
                      <img src={data.data[id].prof_ico} alt="" />
                    </div>
                    <div className="content">
                      <p>
                        <span className="txt-b txt-name">
                          {data.data[id].username}
                        </span>
                        <span>commented {data.data[id].text}</span>
                      </p>
                      <p className="time">{data.data[id].created}</p>
                    </div>
                  </div>
                </Link>
              ))}
            </div>
          ) : null}

          <div className="footer">
            <Link to="/notifications" target="_blank" rel="noopener noreferrer">
              See all notifications
            </Link>
          </div>
        </div>
        <div className="overlay" onClick={() => setOpen(false)}></div>
      </div>
    </div>
  );
}

// Defining the prop types for the component
Notifications.propTypes = {
  theme: PropTypes.oneOf(["light", "dark"]).isRequired,
  variant: PropTypes.oneOf(["primary", "success", "warning", "danger"]),
  markAllRead: PropTypes.func,
};

Notifications.defaultProps = {
  theme: "primary",
  variant: "light",
};

export default Notification;
