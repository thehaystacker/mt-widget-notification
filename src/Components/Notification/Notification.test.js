import React from "react";
import { shallow } from "enzyme";
import jest from "jest-mock";

import Notification from "./index";
import data from "../../Store/Notifications/data";

describe("Notification", () => {
  it("component should render without crashing", () => {
    const component = shallow(<Notification data={data} />);
    expect(component).toMatchSnapshot();
  });

  it("mark all read button will mark all notifications as read", () => {
    const cbClick = jest.fn();
    const component = shallow(
      <Notification data={data} markAllRead={cbClick} />
    );
    component.find('[data-testid="btn-markallread"]').simulate("click");
    expect(cbClick).toHaveBeenCalled();
  });

  it("mark all read function gets called only once", () => {
    const cbClick = jest.fn();
    const component = shallow(
      <Notification data={data} markAllRead={cbClick} />
    );
    component.find('[data-testid="btn-markallread"]').simulate("click");
    expect(cbClick).toHaveBeenCalledTimes(1);
  });
});
