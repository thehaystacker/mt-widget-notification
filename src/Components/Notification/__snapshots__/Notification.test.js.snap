// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component should render without crashing 1`] = `
<div
  className="wrap-notif   "
>
  <div
    className="wrap-ico"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="ico"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f0f3",
            "M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z",
          ],
          "iconName": "bell",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <div
      className="wrap-badge"
    >
      5
    </div>
  </div>
  <div
    className="wrap-dropdown"
  >
    <div
      className="list-box"
    >
      <div
        className="header"
      >
        <FontAwesomeIcon
          border={false}
          className="ico-header bell"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f0f3",
                "M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z",
              ],
              "iconName": "bell",
              "prefix": "far",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h5
          className="txt-h5"
        >
          Notifications
        </h5>
        <FontAwesomeIcon
          border={false}
          className="ico-header read-all link-ico"
          data-testid="btn-markallread"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f058",
                "M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z",
              ],
              "iconName": "check-circle",
              "prefix": "far",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title="Mark all as read"
          transform={null}
        />
      </div>
      <div
        className="list"
      >
        <Link
          key="1010"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1010"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0A816QQ7i-w2DLwWBdY0nmq8-lHCUlcUHBw&usqp=CAU"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Dan Auerbach
                </span>
                <span>
                  commented 
                  Lorem ipsum dolor sit amet, consectetur adipiscing
                </span>
              </p>
              <p
                className="time"
              >
                15 September 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1011"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1011"
        >
          <div
            className="item unread"
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://i0.wp.com/worldbusinessfitness.com/wp-content/uploads/2018/06/opulent-profile-square-02.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Mia Brian
                </span>
                <span>
                  commented 
                  sed do eiusmod tempor incididunt ut labore et dolore
                </span>
              </p>
              <p
                className="time"
              >
                8 October 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1012"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1012"
        >
          <div
            className="item unread"
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://www.nzherald.co.nz/resizer/qsciwSumABgKMvLjKYvh2TK1sGs=/360x384/filters:quality(70)/arc-anglerfish-syd-prod-nzme.s3.amazonaws.com/public/QZI5SNWLCJANLBVQHUDCEQTNTE.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Emily Broemel
                </span>
                <span>
                  commented 
                  minim veniam, quis nostrud exercitation
                </span>
              </p>
              <p
                className="time"
              >
                20 October 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1013"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1013"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQGfF_3ycXBYHLIC4NWYcovKN8W_dsy98dqQw&usqp=CAU"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Dan Auerbach
                </span>
                <span>
                  commented 
                  Excepteur sint occaecat cupidatat non proident
                </span>
              </p>
              <p
                className="time"
              >
                18 April 2012
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1014"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1014"
        >
          <div
            className="item unread"
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://uilove.in/realestate/listo/preview/img/demo/profile.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Alex Lifeson
                </span>
                <span>
                  commented 
                  Lorem ipsum dolor sit amet, consectetur adipiscing
                </span>
              </p>
              <p
                className="time"
              >
                15 September 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1015"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1015"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0A816QQ7i-w2DLwWBdY0nmq8-lHCUlcUHBw&usqp=CAU"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Dan Auerbach
                </span>
                <span>
                  commented 
                  Lorem ipsum dolor sit amet, consectetur adipiscing
                </span>
              </p>
              <p
                className="time"
              >
                15 September 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1016"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1016"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://i0.wp.com/worldbusinessfitness.com/wp-content/uploads/2018/06/opulent-profile-square-02.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Mia Brian
                </span>
                <span>
                  commented 
                  sed do eiusmod tempor incididunt ut labore et dolore
                </span>
              </p>
              <p
                className="time"
              >
                8 October 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1017"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1017"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://www.eventiz.biz/wp-content/uploads/2015/12/speaker-3.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Carl Broemel
                </span>
                <span>
                  commented 
                  minim veniam, quis nostrud exercitation
                </span>
              </p>
              <p
                className="time"
              >
                20 October 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1018"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1018"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQGfF_3ycXBYHLIC4NWYcovKN8W_dsy98dqQw&usqp=CAU"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Dan Auerbach
                </span>
                <span>
                  commented 
                  Excepteur sint occaecat cupidatat non proident
                </span>
              </p>
              <p
                className="time"
              >
                18 April 2012
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1019"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1019"
        >
          <div
            className="item unread"
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://uilove.in/realestate/listo/preview/img/demo/profile.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Alex Lifeson
                </span>
                <span>
                  commented 
                  Lorem ipsum dolor sit amet, consectetur adipiscing
                </span>
              </p>
              <p
                className="time"
              >
                15 September 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1020"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1020"
        >
          <div
            className="item unread"
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0A816QQ7i-w2DLwWBdY0nmq8-lHCUlcUHBw&usqp=CAU"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Dan Auerbach
                </span>
                <span>
                  commented 
                  Lorem ipsum dolor sit amet, consectetur adipiscing
                </span>
              </p>
              <p
                className="time"
              >
                15 September 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1021"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1021"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://i0.wp.com/worldbusinessfitness.com/wp-content/uploads/2018/06/opulent-profile-square-02.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Mia Brian
                </span>
                <span>
                  commented 
                  sed do eiusmod tempor incididunt ut labore et dolore
                </span>
              </p>
              <p
                className="time"
              >
                8 October 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1022"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1022"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://www.eventiz.biz/wp-content/uploads/2015/12/speaker-3.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Carl Broemel
                </span>
                <span>
                  commented 
                  minim veniam, quis nostrud exercitation
                </span>
              </p>
              <p
                className="time"
              >
                20 October 2021
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1023"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1023"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQGfF_3ycXBYHLIC4NWYcovKN8W_dsy98dqQw&usqp=CAU"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Dan Auerbach
                </span>
                <span>
                  commented 
                  Excepteur sint occaecat cupidatat non proident
                </span>
              </p>
              <p
                className="time"
              >
                18 April 2012
              </p>
            </div>
          </div>
        </Link>
        <Link
          key="1024"
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications/1024"
        >
          <div
            className="item "
          >
            <div
              className="prof-ico"
            >
              <img
                alt=""
                src="https://uilove.in/realestate/listo/preview/img/demo/profile.jpg"
              />
            </div>
            <div
              className="content"
            >
              <p>
                <span
                  className="txt-b txt-name"
                >
                  Alex Lifeson
                </span>
                <span>
                  commented 
                  Lorem ipsum dolor sit amet, consectetur adipiscing
                </span>
              </p>
              <p
                className="time"
              >
                15 September 2021
              </p>
            </div>
          </div>
        </Link>
      </div>
      <div
        className="footer"
      >
        <Link
          rel="noopener noreferrer"
          target="_blank"
          to="/notifications"
        >
          See all notifications
        </Link>
      </div>
    </div>
    <div
      className="overlay"
      onClick={[Function]}
    />
  </div>
</div>
`;
